src/__test__/calculate.test.js

```diff
@@ -0,0 +1,278 @@
+import calculate from '../logic/calculate';
+
+describe('Test for the button AC', () => {
+  test('it should return Obj with null values', () => {
+    const obj = {
+      total: 5,
+      next: 2,
+      operation: '+',
+    };
+    const btnValue = 'AC';
+
+    const response = calculate(obj, btnValue);
+
+    expect(response.total).toBe(null);
+    expect(response.next).toBe(null);
+    expect(response.operation).toBe(null);
+  });
+});
+
+describe('Test for button with type of number ', () => {
+  test('case 1: total and operation are null', () => {
+    const obj = {
+      total: null,
+      next: '0',
+      operation: null,
+    };
+    const btnValue = '0';
+    const response = calculate(obj, btnValue);
+    expect(response).toEqual({});
+  });
+});
+
+describe('Test if there is an operation ', () => {
+  test('case 1: If there is an operation, update next which has a value', () => {
+    const obj = {
+      total: 2,
+      next: 4,
+      operation: '+',
+    };
+    const btnValue = '1';
+    // expect(obj.next).toEqual(4);
```

Comment:
- [OPTIONAL] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.